l10n_ro_account_spv/models/message_spv.py

```diff
@@ -0,0 +1,357 @@
+# Copyright (C) 2024 Dorin Hongu <dhongu(@)gmail(.)com
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import io
+import logging
+import zipfile
+from base64 import b64encode
+
+import requests
+from lxml import etree
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class MessageSPV(models.Model):
+    _name = "l10n.ro.message.spv"
+    _description = "Message SPV"
+    _order = "date desc"
+
+    name = fields.Char(string="Message ID")  # id
+    cif = fields.Char(string="CIF")  # cif
+    message_type = fields.Selection(
+        [
+            ("in_invoice", "In Invoice"),
+            ("out_invoice", "Out Invoice"),
+            ("error", "Error"),
+        ],
+        string="Type",
+    )  # tip
+    date = fields.Datetime(string="Date")  # data_creare
+    details = fields.Char(string="Details")  # detalii
+    error = fields.Text(string="Error")  # eroare
+    request_id = fields.Char(string="Request ID")  # id_solicitare
+    ref = fields.Char(string="Reference")  # referinta
+
+    # campuri suplimentare
+
+    invoice_id = fields.Many2one("account.move", string="Invoice")
+    partner_id = fields.Many2one("res.partner", string="Partner")
+
+    # draft - starea initiala a mesajului descarcat din SPV
+    # downloaded - fisierul a fost descarcat cu succes
+    # invoice - factura a fost creata cu succes
+    # done - factura a fost creata si validata cu succes
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("downloaded", "Downloaded"),
+            ("invoice", "Invoice"),
+            ("error", "Error"),
+            ("done", "Done"),
+        ],
+        string="State",
+        default="draft",
+    )
+    file_name = fields.Char(string="File Name")
+    attachment_id = fields.Many2one("ir.attachment", string="Attachment")
+    attachment_xml_id = fields.Many2one("ir.attachment", string="XML")
+    attachment_anaf_pdf_id = fields.Many2one("ir.attachment", string="ANAF PDF")
```

Comment:
Nu ajunge 1 camp, daca e embedded e ala, daca nu cel de anaf

Reply:
parerea mea ca nu e necesar.
cred ca cel de la ANAF e obligatoriu, ca ala e parsat din XML validat. PDF embeded suna mai bine, e mai frumos, dar cred ca e facultativ. Factura fiscala e XML deci randarea PDF din XML cu sigiliu ANAF ar trebui sa fie varianta oficiala 100%
PDF embeded va fi prezent inevitabil ca atasament la factura dupa ce e parsat XML.

Reply:
daca vrei sa verifici ce este in XML